dotCMS/src/integration-test/java/com/dotcms/publishing/job/SiteSearchJobImplTest.java

```diff
@@ -439,74 +445,220 @@ public void Test_Incremental_Job_Test_Pages_Are_Found_Create_And_Publish_New_Pag
+                .defaultPageToDefaultLanguage(true)
+                .siteSearchDefaultLanguage(true)
+                .siteSearchSecondLanguage(true)
+                .createContentInDefaultLanguage(true)
+                .createContentInSecondLanguage(true)
+                .createPageInDefaultLanguage(true)
+                .expectedResultsWhenSearchingContentInDefaultLanguage(true)
+                .expectedResultsWhenSearchingContentInSecondLanguage(true)
+                .createTestCaseSiteSearch();
+
+
+        /*
+         * Given sceneario: two-language (second and third) content referenced from a page in default lang only. DEFAULT_PAGE_TO_DEFAULT_LANGUAGE=true.
+         * Create a Site-Search run including both languages
+         * Expected: searching content of either version of the content should give results. The result should be the page in the default language
+         */
+
+        TestCaseSiteSearch case7 = new TestCaseSiteSearchBuilder()
+                .defaultContentToDefaultLanguage(true)
+                .defaultPageToDefaultLanguage(true)
+                .siteSearchSecondLanguage(true)
+                .siteSearchThirdLanguage(true)
+                .createContentInSecondLanguage(true)
+                .createContentInThirdLanguage(true)
+                .createPageInDefaultLanguage(true)
+                .expectedResultsWhenSearchingContentInSecondLanguage(true)
+                .expectedResultsWhenSearchingContentInThirdLanguage(true)
+                .createTestCaseSiteSearch();
+
+
+        return new TestCaseSiteSearch[] {case1, case2, case3, case4, case5, case6, case7};
+    }
 
+    @UseDataProvider("siteSearchTestCases")
     @Test
-    public void test_MultilangContent_IndexingAllLanguages()
+    public void testSiteSearchDifferentScenarios(final TestCaseSiteSearch testCase)
             throws DotPublishingException, JobExecutionException, DotDataException, IOException, DotSecurityException, WebAssetException {
 
         boolean defaultContentToDefaultLangOriginalValue =
                 Config.getBooleanProperty("DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE", false);
 
-        try {
+        boolean defaultPagetoDefaultLangOriginalValue =
+                Config.getBooleanProperty("DEFAULT_PAGE_TO_DEFAULT_LANGUAGE", true);
 
-            Config.setProperty("DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE", true);
+        try {
+            Config.setProperty("DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE", testCase.defaultContentToDefaultLanguage);
+            Config.setProperty("DEFAULT_PAGE_TO_DEFAULT_LANGUAGE", testCase.defaultPageToDefaultLanguage);
 
             final Host site = new SiteDataGen().nextPersisted();
-            Language lang1 = APILocator.getLanguageAPI().getDefaultLanguage();
-            Language lang2 = new LanguageDataGen().nextPersisted();
+            Language defaultLang = APILocator.getLanguageAPI().getDefaultLanguage();
+            Language secondLang = new LanguageDataGen().nextPersisted();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'secondLang' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5303548)